algorithms/src/snark/marlin/ahp/prover/round_functions/second.rs

```diff
@@ -98,7 +98,7 @@ impl<F: PrimeField, MM: MarlinMode> AHPForR1CS<F, MM> {
 
         let sumcheck_lhs = Self::calculate_lhs(&state, t, summed_z_m, z, *alpha);
 
-        debug_assert!(
+        assert!(
```

Comment:
note: this was presumably a `debug_assert` for performance reasons; might make sense to revert back to it, unless `evaluate_over_domain_by_ref` became fast enough in the meantime